packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -285,6 +285,43 @@ export class JavaBuilder implements TargetBuilder {
 
     const profileName = 'local-jsii-modules';
     const localRepository = this.options.arguments['maven-local-repository'];
+
+    const repos = localRepos.map((repo) => ({
+      id: repo.replace(/[\\/:"<>|?*]/g, '$'),
+      url: `file://${repo}`,
+    }));
+
+    let servers = undefined;
+
+    // Add repository provided by user to support edge cases
+    // e.g. corporate proxies and other remote repositories
+    const extraRepositoryId: string | undefined =
+      this.options.arguments['maven-extra-repository-id'];
+    const extraRepositoryUrl: string | undefined =
+      this.options.arguments['maven-extra-repository-url'];
+    const extraRepositoryUsername: string | undefined =
+      this.options.arguments['maven-extra-repository-username'];
+    const extraRepositoryPassword: string | undefined =
+      this.options.arguments['maven-extra-repository-password'];
+
+    if (extraRepositoryId) {
+      if (!extraRepositoryUrl)
+        throw new Error(
+          'Extra repository requested but no URL was provided! Use the --maven-extra-repository-url argument to provide a value.',
+        );
```

Comment:
Could this maybe be made a single option? Something like `--maven-extra-repository=id=url` or if you need credentials (password caveat from above still applies) `--maven-extra-repository=id=username:password@url` (although maybe `yargs` won't like this format?).

Then - is there a use-case for having more than 1 extra repository configured?